zbench.zig

```diff
@@ -274,6 +274,69 @@ pub fn getSystemInfo() !platform.OsInfo {
     return try platform.getSystemInfo();
 }
 
+/// A collection of the results of each benchmark. The results are available in
+/// the `results` array but they can be collectively pretty printed and
+/// deallocated with this structure.
+pub const Results = struct {
+    allocator: std.mem.Allocator,
+    results: []Result,
+
+    pub fn deinit(self: Results) void {
+        for (self.results) |r| r.deinit();
+        self.allocator.free(self.results);
+    }
+
+    /// Formats and prints the benchmark results in a human readable format.
+    /// writer: Type that has the associated method print (for example std.io.getStdOut.writer())
+    /// colors: Whether to pretty-print with ANSI colors or not.
+    pub fn prettyPrint(self: Results, writer: anytype, colors: bool) !void {
+        try format.prettyPrintHeader(writer);
+        for (self.results) |r| try r.prettyPrint(writer, colors);
+    }
+
+    /// Prints a summary output at the end of benchmarking sessions
+    /// This summary highlights the fastest benchmark by name in green, and compares each
+    /// subsequent benchmark to show how many times slower they are relative to the fastest.
+    pub fn printSummary(self: Results, writer: anytype) !void {
```

Comment:
I removed the `Results` struct in the refactor, I wasn't thinking about this branch sorry. Maybe `printSummary()` could take a parameter like `statistics: []struct {[]const u8, Statistics(u64)}` instead of `self: Results`? I don't think it needs anything besides the name and the stats.